app/controllers/admin/dashboards_controller.rb

```diff
@@ -1,7 +1,27 @@
 module Admin
   class DashboardsController < AdminController
-    def show
-      @buildings = Building.all.ordered_by_uprn
+    before_action :fetch_buildings, :filtered?, :fetch_total
+
+    def show; end
+
+    private
+
+    def fetch_buildings
+      scope = Building.left_outer_joins(:surveys)
+
+      scope = scope.where.not('surveys.completed_at' => nil) if params[:completed] == "1"
+
+      scope = scope.where('surveys.completed_at' => nil) if params[:not_received] == "1"
+
+      @buildings = scope
+    end
```

Comment:
Any reason not to pull in the `Browseable` concern from Petitions? It has support for filters as well even though it's not used heavily:

https://github.com/alphagov/e-petitions/blob/3363b21a2e803ebb7fc19891835cf5f1f0fa079a/app/models/petition.rb#L82